src/Domain/Python/Block.cpp

```diff
@@ -0,0 +1,35 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Domain/Python/Block.hpp"
+
+#include <cstddef>
+#include <pybind11/pybind11.h>
+#include <pybind11/stl.h>
+#include <string>
+#include <vector>
+
+#include "Domain/Block.hpp"
+#include "Utilities/GetOutput.hpp"
+
+namespace py = pybind11;
+
+namespace domain::py_bindings {
+
+namespace {
+template <size_t Dim>
+void bind_block_impl(py::module& m) {  // NOLINT
+  py::class_<Block<Dim>>(m, ("Block" + get_output(Dim) + "D").c_str())
+      .def("is_time_dependent", &Block<Dim>::is_time_dependent)
+      .def_property_readonly("id", &Block<Dim>::id)
+      .def_property_readonly("name", &Block<Dim>::name);
```

Comment:
Add method to get maps? Up to you :)

Reply:
depends on #4630

Reply:
With #4630 merged I just rebased and added this now.